pkg/containertools/imagereader.go

```diff
@@ -150,52 +153,28 @@ func extractBundleManifests(layerTarball, outputDir string, tarReader *tar.Reade
 			return err
 		}
 
-		if header.Typeflag == tar.TypeReg {
-			if header.Name == layerTarball {
-				// Found the embedded tarball for the layer
-				layerReader := tar.NewReader(tarReader)
-
-				err = extractTarballToDir(outputDir, layerReader)
-				if err != nil {
-					return err
-				}
-
-				return nil
+		if header.Typeflag == tar.TypeReg && header.Name == layerTarball {
+			// Found the embedded tarball for the layer
+			decompressed, err := compression.DecompressStream(tarReader)
+			if err != nil {
+				return err
 			}
-		}
-	}
-}
+			_, err = archive.Apply(context.TODO(), outputDir, decompressed, archive.WithFilter(adjustPerms))
 
-func extractTarballToDir(outputDir string, tarReader *tar.Reader) error {
-	for {
-		header, err := tarReader.Next()
-		if err != nil {
-			if err == io.EOF {
-				return nil
-			}
 			return err
 		}
+	}
+}
 
-		switch header.Typeflag {
-		case tar.TypeDir:
-			// Create the directory if it doesn't exist
-			directoryToWrite := filepath.Join(outputDir, header.Name)
-			if _, err := os.Stat(directoryToWrite); os.IsNotExist(err) {
-				os.Mkdir(directoryToWrite, 0777)
-			}
-		case tar.TypeReg:
-			manifestToWrite := filepath.Join(outputDir, header.Name)
+func adjustPerms(h *tar.Header) (bool, error) {
```

Comment:
this happens in two places now, right? should this be shared so that we unpack things the same way every time?

Reply:
I think so. I expect us to move this logic into the `image` package in the future. I didn't know if doing that in a bug-fix would be overkill for such a small amount of code.